neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -538,11 +538,18 @@ def update_invocation_context(self, invocation_context: InvocationContext):
         Update internal state based on the InvocationContext instance passed in.
         :param invocation_context: The context policy container that pertains to the invocation
         """
+        old_interceptor: InterceptingJournal = self.interceptor
         self.invocation_context = invocation_context
 
         # Make a nested chain where each journal is wrapped by the next
         base_journal: Journal = self.invocation_context.get_journal()
         self.interceptor = InterceptingJournal(wrapped_journal=base_journal, origin=self.origin)
+
+        # The SystemMessage has already been written to the journal
+        # need to transfer it over when this shift happens.
+        if old_interceptor is not None:
+            for message in old_interceptor.get_messages():
+                self.interceptor.write_unwrapped_message(message, self.origin)
```

Comment:
This fixes the system message going to the tracing output.
There is a weird race condition I don't really have my head around fully where this update_invocation_context() method gets called more than once, notably *after* the first SystemMessage has been written out.
The Interceptor needs all the history to properly report the tracing.